### DISCUSSION TITLE
Dcm4chee-arc secured access Retrieve URL link got "Unauthorized" error

### DISCUSSION THREAD

**Discussion by oliverfchen** [dcm4che/dcm4chee-arc-light#3360] [Q&A] [answered]
In secured dcm4chee-arc, can't download dcm using Retrieve URL (0008,1190) :
https://192.168.8.199:8443/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.2.276.0.7230010.3.1.4.1503774336.864.1487142125.652/series/1.2.276.0.7230010.3.1.4.1503774336.864.1487142092.651/instances/1.2.276.0.7230010.3.1.4.1503774336.864.1487142212.654
return "Unauthorized" error.
How to fix please?

**Comment by gunterze** [comment, marked as answer]
You have to pass a valid OpenID Connect Access Token (fetched from Keycloak) in Query Parameter `access_token` or (better) in HTTP `Authorization` header.

s. [Getting OIDC Access Token using curl](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Getting-OIDC-Access-Token-using-curl)

**Reply by oliverfchen** [reply]
Thanks, but my screen is different from the tutorial page, it doesn't have "Service Account Roles" and "Credentials" taps. I am using the lastest Keycloak image 11.0.3.

**Reply by gunterze** [reply]
Verify `Service Accounts Enabled = ON`

**Reply by oliverfchen** [reply]
Sorry, I have "Enabled", "Standard Flow Enabled", "Implicit Flow Enabled" and "Direct Access Grants Enabled", but don't have the "Service Accounts Enabled", or "Credentials" tab.

**Reply by gunterze** [reply]
Verify `Access Type: confidential`

**Reply by oliverfchen** [reply]
YES! Got that, thank you very much!
